spec/patterns/shorewall_spec.rb

```diff
@@ -2,89 +2,101 @@
+                                "ttl"=>63,
+                                "id"=>"55251"
+                            },
+                            "network"=>{"transport"=>"UDP"},
+                            "source"=>{"ip"=>"1.2.3.4", "port"=>5353},
+                            "destination"=>{"ip"=>"192.168.0.10", "port"=>5335}
+                        )
+      else
+        should include("nf_host" => "myHost")
+        should include("nf_action1" => "FORWARD")
+        should include("nf_action2" => "REJECT")
+        should include("nf_in_interface" => "eth2")
+        should include("nf_out_interface" => "eth2")
+        should include("nf_src_ip" => "1.2.3.4")
+        should include("nf_dst_ip" => "192.168.0.10")
+        should include("nf_len" => "141")
+        should include("nf_tos" => "0x00")
+        should include("nf_prec" => "0x00")
+        should include("nf_ttl" => "63")
+        should include("nf_id" => "55251")
+        should include("nf_protocol" => "UDP")
+        should include("nf_src_port" => "5353")
+        should include("nf_dst_port" => "5335")
+      end
+    end
 
-    it { should include("nf_dst_port" => "5353") }
   end
 
   context "parsing a message without OUT interface" do
 
-    let(:value) { "May 28 17:31:07 myHost kernel: [3125121.106700] Shorewall:net2fw:DROP:IN=eth1 OUT= MAC=00:02:b3:c7:2f:77:38:72:c0:6e:92:9c:08:00 SRC=1.2.3.4 DST=1.2.3.4 LEN=60 TOS=0x00 PREC=0x00 TTL=49 ID=6480 DF PROTO=TCP SPT=59088 DPT=8080 WINDOW=2920 RES=0x00 SYN URGP=0" }
-
-    subject     { grok_match(pattern, value) }
-
-    it { should include("timestamp" => "May 28 17:31:07") }
-
-    it { should include("nf_host" => "myHost") }
-
-    it { should include("nf_action1" => "net2fw") }
-
-    it { should include("nf_action2" => "DROP") }
-
-    it { should include("nf_in_interface" => "eth1") }
-
-    it { expect(subject["nf_out_interface"]).to be_nil }
-
-    it { should include("nf_dst_mac" => "00:02:b3:c7:2f:77") }
-
-    it { should include("nf_src_mac" => "38:72:c0:6e:92:9c") }
-
-    it { should include("nf_src_ip" => "1.2.3.4") }
-
-    it { should include("nf_dst_ip" => "1.2.3.4") }
-
-    it { should include("nf_len" => "60") }
-
-    it { should include("nf_tos" => "0x00") }
-
-    it { should include("nf_prec" => "0x00") }
-
-    it { should include("nf_ttl" => "49") }
-
-    it { should include("nf_id" => "6480") }
-
-    it { should include("nf_protocol" => "TCP") }
 
-    it { should include("nf_src_port" => "59088") }
+    let(:message) do
+      "May 28 17:31:07 server Shorewall:net2fw:DROP:" +
+      "IN=eth1 OUT= MAC=00:02:b3:c7:2f:77:38:72:c0:6e:92:9c:08:00 SRC=127.0.0.1 DST=1.2.3.4 LEN=60 TOS=0x00 PREC=0x00 TTL=49 ID=6480 DF PROTO=TCP SPT=59088 DPT=8080 WINDOW=2920 RES=0x00 SYN URGP=0"
+    end
+
+    it 'matches' do
+      should include("timestamp" => "May 28 17:31:07")
+      if ecs_compatibility?
+        expect(grok).to include(
```

Comment:
nit: using [rspec's `aggregate_failures`](https://relishapp.com/rspec/rspec-expectations/docs/aggregating-failures) can help tell the whole story a lot better in cases where we have many unchained expectations about the result of a single action.

Reply:
:bow: interesting feature, not really an rspec person I just suffer my way through specs :wink: 
in this particular case I do like the failures one-by-one since there likely will be multiple cases failing due the same cause already, as the whole file is run. these specs could have been much better have I spent more time writing a few helpers...

thanks for the hint again, will consider using this feature.